### ISSUE TITLE
Option to backup and restore from gcp storage bucket (GCS)

### ISSUE THREAD

**Issue by devopsevd** [CrunchyData/postgres-operator#1791]
**What is the motivation or use case for the change? **
I do not see an option using the crunchydata operator to backup and restore from GCP storage bucket.

**Describe the solution you'd like**
I would to see an option while using the postgres-operator to backup and restore from GCP storage bucket.

**Please tell us about your environment:**

* Operating System: Linux based 
* Where is this running ( Local, Cloud Provider) - GCP GKE cluster
* Storage being used (NFS, Hostpath, Gluster, etc):GCP PVC, 
* Container Image Tag: centos7-4.3.0 / centos7-12.2-4.3.0
* PostgreSQL Version:12.2
* Platform (Docker, Kubernetes, OpenShift): Kubernetes
* Platform Version: 1.16.11-gke.5

**Additional context**
Add any other context or screenshots about the enhancement request here.

**Comment by jkatz** [comment]
For direct support of GCS, the PostgreSQL Operator will need for [pgBackRest](https://www.pgbackrest.org) to support GCS.

Until then, I would recommend reviewing [this comment](https://github.com/CrunchyData/postgres-operator/issues/1761#issuecomment-670215596) about how GCS can be accomplished with MinIO today, substituting "Azure" with GCS for the supported gateway:

[https://docs.min.io/docs/minio-gateway-for-gcs.html](https://docs.min.io/docs/minio-gateway-for-gcs.html)

**Comment by davi5e** [comment]
Hello,

I just tried to enable the backups in GCS using [HMAC keys for service accounts](https://cloud.google.com/storage/docs/authentication/managing-hmackeys) as it should be S3 compatible but it would appear I just hit this issue where [it's not fully supported](https://github.com/minio/mc/issues/3073#issuecomment-591877661).

It would appear that some small changes in PGO _would_ seamlessly enable this feature and I do use this approach elsewhere.

The actual error stanza logs is shown below. Just to be clear, this is an "error" on Google's API AFAIK...

```
time="2020-08-30T22:58:15Z" level=info msg="pgo-backrest starts"
time="2020-08-30T22:58:15Z" level=info msg="debug flag set to false"
time="2020-08-30T22:58:15Z" level=info msg="backrest stanza-create command requested"
time="2020-08-30T22:58:15Z" level=info msg="s3 flag enabled for backrest command"
time="2020-08-30T22:58:15Z" level=info msg="command to execute is [pgbackrest stanza-create  --db-host=10.3.4.10 --db-path=/pgdata/website-cluster --repo1-type=s3 --no-repo1-s3-verify-tls]"
time="2020-08-30T22:58:15Z" level=info msg="command is pgbackrest stanza-create  --db-host=10.3.4.10 --db-path=/pgdata/website-cluster --repo1-type=s3 --no-repo1-s3-verify-tls "
time="2020-08-30T22:58:19Z" level=error msg="command terminated with exit code 39"
time="2020-08-30T22:58:19Z" level=info msg="output=[]"
time="2020-08-30T22:58:19Z" level=info msg="stderr=[ERROR: [039]: S3 request failed with 400: Bad Request\n       *** URI/Query ***:\n       /?delimiter=%2F&list-type=2&prefix=backrestrepo%2Fwebsite-cluster-backrest-shared-repo%2Farchive%2Fdb%2F\n       *** Request Headers ***:\n       authorization: <redacted>\n       content-length: 0\n       host: my-bucket-name-here.storage.googleapis.com\n       x-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\n       x-amz-date: <redacted>\n       *** Response Headers ***:\n       cache-control: private, max-age=0\n       content-length: 239\n       content-type: application/xml; charset=UTF-8\n       date: Sun, 30 Aug 2020 22:58:19 GMT\n       expires: Sun, 30 Aug 2020 22:58:19 GMT\n       server: UploadServer\n       x-guploader-uploadid: ABg5-UyZgtOk7vQuoGHtbE6G2nxVFheoWpmh8GV1R0_G7KqT8hcJocA1_768nQLPgaXuxxFLxYyvkFkWqNxSHgnN08BInmyXBA\n       *** Response Content ***:\n       <?xml version='1.0' encoding='UTF-8'?><Error><Code>NotImplemented</Code><Message>A header or query you provided requested a function that is not implemented.</Message><Details>GET ?list-type is not implemented for buckets</Details></Error>\n]"
time="2020-08-30T22:58:19Z" level=error msg="command terminated with exit code 39"
stream closed
```

**Comment by strigona-worksight** [comment]
Just FYI, the stanza logs indicates the issue at hand with GCS:
>GET ?list-type is not implemented for buckets

GCS is not fully compatible with S3 protocol. pgbackrest needs to be modified to support the storage endpoint and this feature is on pgbackest's backlog: https://github.com/pgbackrest/pgbackrest/projects/2#card-33945748

**Comment by leosussan** [comment]
Similar problem for me. I've been experimenting with an implementation that uses a GCS CSI driver to mount a Cloud Storage bucket as a persistent volume:
https://github.com/ofek/csi-gcs

Mileage may vary, but it works well for me.

**Comment by jkatz** [comment]
Direct support for GCS will be available with the 4.7.0 release.

**Comment by kubaracek** [comment]
@jkatz Is there any ETA on 4.7.0? I would like to play around with the GCS and put it on our servers at some point so am thinking if I should wait for the official release or play with master for now. Thank you
